frontend/src/metabase/actions/components/ActionForm/ActionFormFieldWidget.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { forwardRef } from "react";
+import _ from "underscore";
+
+import type {
+  ActionFormFieldProps,
+  InputComponentType,
+} from "metabase-types/api";
+
+import FormInputWidget from "metabase/core/components/FormInput";
+import FormTextAreaWidget from "metabase/core/components/FormTextArea";
+import FormRadioWidget from "metabase/core/components/FormRadio";
+import FormSelectWidget from "metabase/core/components/FormSelect";
+import FormNumericInputWidget from "metabase/core/components/FormNumericInput";
+import FormBooleanWidget from "metabase/core/components/FormToggle";
+import CategoryFieldPicker from "metabase/components/FormCategoryInput";
+
+const VerticalRadio = (props: any) => <FormRadioWidget {...props} vertical />;
+
+const WIDGETS: Record<InputComponentType, React.FunctionComponent<any>> = {
+  text: FormInputWidget,
+  date: FormInputWidget,
+  time: FormInputWidget,
+  "datetime-local": FormInputWidget,
```

Comment:
btw why don't you want to use `FormDateInput` here?

Reply:
we certainly should. https://github.com/metabase/metabase/issues/27387